src/com/goide/ui/GoSettingsConfigurable.java

```diff
@@ -0,0 +1,140 @@
+import com.intellij.uiDesigner.core.Spacer;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+
+import static com.goide.GoConstants.GO_ONSAVE_FMT_KEY;
+import static com.goide.GoConstants.GO_ONSAVE_IMPORTS_KEY;
+
+public class GoSettingsConfigurable implements SearchableConfigurable {
+  private JRadioButton doNothingOnSave;
+  private JRadioButton goFmtOnSave;
+  private JRadioButton goimportsOnSave;
+  private ButtonGroup myButtonGroup = new ButtonGroup();
+  private final JPanel myPanel;
+
+  public GoSettingsConfigurable() {
+    doNothingOnSave = new JRadioButton("Do nothing");
+    doNothingOnSave.setMnemonic('n');
+    goFmtOnSave = new JRadioButton("go fmt", true);
+    goFmtOnSave.setMnemonic('f');
+    goimportsOnSave = new JRadioButton("goimports");
+    goimportsOnSave.setMnemonic('i');
+
+    myButtonGroup.add(doNothingOnSave);
+    myButtonGroup.add(goFmtOnSave);
+    myButtonGroup.add(goimportsOnSave);
+
+    final GridLayoutManager layoutManager = new GridLayoutManager(4, 1, new Insets(0, 0, 0, 0), -1, -1);
+    myPanel = new JPanel(layoutManager);
+    final Spacer spacer = new Spacer();
+    myPanel.add(spacer, new GridConstraints(3, 0, 1, 1, GridConstraints.ANCHOR_NORTHWEST,
+                                              GridConstraints.FILL_NONE, GridConstraints.SIZEPOLICY_FIXED,
+                                              GridConstraints.SIZEPOLICY_FIXED, null, null, null));
+
+    myPanel.add(doNothingOnSave, buttonConstrains(0));
+    myPanel.add(goFmtOnSave, buttonConstrains(1));
+    myPanel.add(goimportsOnSave, buttonConstrains(2));
+  }
+
+  @NotNull
+  @Override
+  public String getId() {
+    return "go";
+  }
+
+  @Override
+  public Runnable enableSearch(String option) {
+    return null;
+  }
+
+  @Nls
+  @Override
+  public String getDisplayName() {
+    return "Go Settings";
+  }
+
+  public Icon getIcon() {
+    return GoIcons.ICON;
+  }
+
+  @Nullable
+  @Override
+  public String getHelpTopic() {
+    return null;
+  }
+
+  @Override
+  public JComponent createComponent() {
+    return myPanel;
+  }
+
+  @Override
+  public boolean isModified() {
+    if (PropertiesComponent.getInstance().getBoolean(GO_ONSAVE_FMT_KEY, true) != goFmtOnSave.isSelected()) {
+      return true;
+    }
+
+    if (PropertiesComponent.getInstance().getBoolean(GO_ONSAVE_IMPORTS_KEY, false) != goimportsOnSave.isSelected()) {
+      return true;
+    }
+
+    return false;
+  }
+
+  @Override
+  public void apply() throws ConfigurationException {
+    PropertiesComponent.getInstance().setValue(GO_ONSAVE_FMT_KEY, String.valueOf(goFmtOnSave.isSelected()));
+    PropertiesComponent.getInstance().setValue(GO_ONSAVE_IMPORTS_KEY, String.valueOf(goimportsOnSave.isSelected()));
+  }
+
+  @Override
+  public void reset() {
+    Boolean goFmt = PropertiesComponent.getInstance().getBoolean(GO_ONSAVE_FMT_KEY, true);
```

Comment:
These properties are not some 'flags' for defining IDEA behaviour. It's real user settings. Please bring back the service that has semantic value and can be exportable.